src/io/output_element.hh

```diff
@@ -0,0 +1,290 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * 
+ * @file    output_element.hh
+ * @brief   Class OutputElement and its iterator OutputElementIterator on the output mesh.
+ */
+
+#ifndef OUTPUT_ELEMENT_HH_
+#define OUTPUT_ELEMENT_HH_
+
+#include "output_mesh.hh"
+
+template <int spacedim>
+class ElementAccessor;
+
+
+#include "mesh/accessors.hh"
+#include "mesh/mesh.h"
+#include "mesh/point.hh"
+#include "output_mesh.hh"
+
+
+/** @brief Represents an element of the output mesh.
+ * Provides element access on the data of the output mesh (nodes, connectivity, offsets etc.).
+ * 
+ * Vertex function suppose spacedim = 3 at the moment.
+ */
+class OutputElement
```

Comment:
I don't like two things in this class:
- duplicated interface for non-disc and disc cases
- creating vertex_list vectors and passing them by value

For the first issue either introduce aux objects for non-disc and disc mesh, usage example:

	output_mesh_->continuous_elements().begin()
	output_mesh_->discontinuous_elements().begin()

Or possibly create whole new  OutputMesh for the discontinus case which can share some date with the continuous mesh due to shared pointers (so e.g. it is safe to delete the continuous mesh).

For the second issue, there are three solutions:
- use our own allocators for those lists in order to share the contents with the big vector
- have our own proxy object for those lists and add new variant of the Field<>::value_list
- use vector of vectors to store element data and create the global arrays only on demand (filling the field values is called  more times then creating the mesh big arrays).